module/solutions/helper.js

```diff
@@ -142,4 +1012,2537 @@ module.exports = class SolutionsHelper {
+
+   static checkForTargetedSolution(
+    link = "",
+    bodyData = {},
+    userId = "",
+    userToken = ""
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let response = {
+          isATargetedSolution: false,
+          link: link,
+        };
+
+        let solutionDetails = await solutionsQueries.solutionsDocument({ link: link }, [
+          "type",
+          "_id",
+          "programId",
+          "name",
+          "projectTemplateId",
+          "programName",
+          "status",
+        ]);
+
+        let queryData = await this.queryBasedOnRoleAndLocation(bodyData);
+        if (!queryData.success) {
+          return resolve(queryData);
+        }
+
+        queryData.data["link"] = link;
+        let matchQuery = queryData.data;
+
+        let solutionData = await solutionsQueries.solutionsDocument(matchQuery, [
+          "_id",
+          "link",
+          "type",
+          "programId",
+          "name",
+          "projectTemplateId",
+        ]);
+
+        if (!Array.isArray(solutionData) || solutionData.length < 1) {
+          response.solutionId = solutionDetails[0]._id;
+          response.type = solutionDetails[0].type;
+          response.name = solutionDetails[0].name;
+          response.programId = solutionDetails[0].programId;
+          response.programName = solutionDetails[0].programName;
+          response.status = solutionDetails[0].status;
+
+          return resolve({
+            success: true,
+            message:
+              CONSTANTS.apiResponses.SOLUTION_NOT_FOUND_OR_NOT_A_TARGETED,
+            result: response,
+          });
+        }
+
+        response.isATargetedSolution = true;
+        Object.assign(response, solutionData[0]);
+        response.solutionId = solutionData[0]._id;
+        response.projectTemplateId = solutionDetails[0].projectTemplateId
+          ? solutionDetails[0].projectTemplateId
+          : "";
+        response.programName = solutionDetails[0].programName;
+        delete response._id;
+
+        return resolve({
+          success: true,
+          message: CONSTANTS.apiResponses.SOLUTION_DETAILS_VERIFIED,
+          result: response,
+        });
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+
+      /**
+      * List of user assigned observations.
+      * @method
+      * @name userAssigned
+      * @param {String} userId - logged in user id.
+      * @param {Number} pageNo - Recent page no.
+      * @param {Number} pageSize - Size of page.
+      * @param {String} search - search text.
+      * @param {String} [ filter = ""] - filter text.
+      * @returns {Object} List of user assigned observations.
+     */
+
+      static assignedObservations(userId, pageNo, pageSize, search,filter = "" ) {
```

Comment:
@Prajwal17Tunerlabs observation function you need to add it in observation helper